blocks/init/src/Blocks/components/paragraph/components/paste-handling.js

```diff
@@ -0,0 +1,136 @@
+        }
+
+        // Insert them one by one and keep spaces
+        for (const element of processedElements) {
+            selectedRange = selectedRange.cloneRange();
+            selectedRange.insertNode(element);
+            selectedRange.setStartAfter(element);
+            selectedRange.collapse(true);
+        }
+    }
+    // ...else append at the end
+    else {
+        setAttributes({ [`${componentName}Content`]: `${attributes.paragraphContent}${parsedParagraphs[0]}` });
+    }
+
+    for (const paragraph of parsedParagraphs.slice(1)) {
+
+        // Create a new block with the same attributes, but different content
+        const block = createBlock(`eightshift-boilerplate/${componentName}`, {
+            ...attributes,
+            [`${componentName}Content`]: paragraph
+        });
+
+        const currentBlockClientId = wp.data.select('core/block-editor').getSelectedBlockClientId();
+
+        const blockIndex = wp.data.select('core/block-editor').getBlockIndex(currentBlockClientId);
+
+        // Insert the block
+        wp.data.dispatch('core/block-editor').insertBlock(block, blockIndex + 1);
+    }
+}
+
+/**
+ * Paste event handler.
+ *
+ * @param {Event} event Passed event parameters.
+ * @param {Object} attributes Component attributes.
+ * @param {Function} setAttributes Component attribute setter.
+ */
+const pasteInto = (event, attributes, setAttributes) => {
+    event.preventDefault();
+    const [copiedData, copiedHtml] = event.clipboardData.items;
+
+    // If HTML content is in clipboard, process that...
+    if (copiedHtml) {
+        copiedHtml.getAsString((inputText) => {
+            // Get all tags from the text
+            const allTags = [...inputText.matchAll(/<.+?>/g)];
+
+            const allowedTags = manifest.pasteAllowTags ?? ['strong', 'b', 'i', 'em', 'span'];
+
+            // Filter out attributes and disallowed tags,
+            // make <p> and <br> elements give newline
+            for (const tag of allTags) {
+                const tagInnardsRegex = /(\S+)=["']?((?:.(?!["']?\s+(?:\S+)=|\s*\/?[>"']))+.)["']?/g;
+                const parsedTag = tag[0].replace(tagInnardsRegex, '');
+                const filteredTagName = parsedTag.replaceAll(/[<>/\s]+/g, '');
+
+                if (allowedTags.includes(filteredTagName)) {
+                    inputText = inputText.replace(tag[0], parsedTag).replace(' >', '>');
+                } else if ((filteredTagName === 'p' && !tag[0].includes('/')) || filteredTagName === 'br') {
+                    inputText = inputText.replace(tag[0], '\n');
+                } else {
+                    inputText = inputText.replace(tag[0], '');
+                }
+            }
+
+            handlePaste(inputText, attributes, setAttributes);
+        });
+    }
+    // ...else handle plaintext
+    else if (copiedData) {
+        copiedData.getAsString((inputText) => handlePaste(inputText, attributes, setAttributes));
+    }
+};
+
+export { pasteInto };
```

Comment:
is this something that is going to change to be project specific or is this going to be always this way in any project?

Reply:
That part should stay the same, it uses configurable options from the manifest (allowed tags) if stuff needs to be changed in a project.

Reply:
then this should go to https://github.com/infinum/eightshift-frontend-libs/tree/develop/scripts/editor and be exported and used directly from the lib and not copied to the project